lib/read.js

```diff
@@ -36,49 +36,55 @@ module.exports = read
  */
 
 function read (req, res, next, parse, debug, options) {
-  var length
-  var opts = options
-  var stream
-
   // read options
-  var encoding = opts.encoding !== null
-    ? opts.encoding
-    : null
-  var verify = opts.verify
-
-  try {
-    // get the content stream
-    stream = contentstream(req, debug, opts.inflate)
-    length = stream.length
-    stream.length = undefined
-  } catch (err) {
-    return next(err)
+  const charset = options.charset
```

Comment:
I renamed the option to `charset` because it holds the `content-type` header charset or the default charset